File: QuanLyPhongKham.Server/src/QuanLyPhongKham.Models/Entities/LICHHEN.cs
using MayNghien.Infrastructures.Models;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace QuanLyPhongKham.Models.Entities
{
    public class LICHHEN : BaseEntity
    {
        [Key]
        public Guid MaLH { get; set; }
        public DateTime ThoiGianKham { get; set; }
        public string TrangThai { get; set; }

        [ForeignKey("BenhNhan")]
        public Guid? MaBN { get; set; }
        public BENHNHAN? BenhNhan { get; set; }

        [ForeignKey("BacSi")]
        public Guid? MaBS { get; set; }
        public BACSI? BacSi { get; set; }

        public PHIEUKHAM? PhieuKham { get; set; }
    }
}

File: QuanLyPhongKham.Client/src/models/LichHen.js
/**
 * @typedef {Object} LichHen
 * @property {string} maLH
 * @property {string|Date} thoiGianKham
 * @property {string} trangThai
 * @property {string} maBN
 * @property {string} maBS
 * @property {string} createdBy
 */

export const createLichHen = (
  maLH = "",
  thoiGianKham = null,
  trangThai = "",
  maBN = null,
  maBS = null,
  createdBy = "",
) => ({
  maLH,
  thoiGianKham,
  trangThai,
  maBN,
  maBS,
  createdBy,
});

export const normalizeLichHen = (item) => ({
  maLH: item?.maLH ?? item?.MaLH ?? "",
  thoiGianKham: item?.thoiGianKham ?? item?.ThoiGianKham ?? null,
  trangThai: item?.trangThai ?? item?.TrangThai ?? "",
  maBN: item?.maBN ?? item?.MaBN ?? null,
  maBS: item?.maBS ?? item?.MaBS ?? null,
  createdBy: item?.createdBy ?? item?.CreatedBy ?? "",
});
